site/src/component/GradeDist/GradeDist.tsx

```diff
@@ -38,24 +38,24 @@ const GradeDist: FC<GradeDistProps> = (props) => {
   const [quarterEntries, setQuarterEntries] = useState<Entry[]>(null!);
 
   const fetchGradeDistData = () => {
-    let url = '';
-    let params = {};
+    let request: Promise<GradesRaw> | undefined;
```

Comment:
is there a reason `request` can be `undefined`? if it's just the `let` statement, then overriding it in all branches should let you define it as just the Promise

Reply:
No, we should be able to remove undefined from the type and just use a non-null assertion below rather than an optional chaining operator